### PR TITLE
tfel: enable python bindings

### PR DESCRIPTION
**Pull request by maxence-wz**
- [x] Have you followed the [guidelines for contributing](https://github.com/Homebrew/homebrew-core/blob/HEAD/CONTRIBUTING.md)?
- [x] Have you ensured that your commits follow the [commit style guide](https://docs.brew.sh/Formula-Cookbook#commit)?
- [x] Have you checked that there aren't other open [pull requests](https://github.com/Homebrew/homebrew-core/pulls) for the same formula update/change?
- [x] Have you built your formula locally with `brew install --build-from-source <formula>`, where `<formula>` is the name of the formula you're submitting?
- [x] Is your test running fine `brew test <formula>`, where `<formula>` is the name of the formula you're submitting?
- [x] Does your build pass `brew audit --strict <formula>` (after doing `brew install --build-from-source <formula>`)? If this is a new formula, does it pass `brew audit --new <formula>`?

-----

### CODE REVIEW

File: Formula/tfel.rb [hunk 1]
```diff
@@ -6,18 +6,10 @@ class Tfel < Formula
   license "GPL-1.0-or-later"
   head "https://github.com/thelfer/tfel.git", using: :git, branch: "master"
 
-  bottle do
-    sha256 arm64_monterey: "76becea0d88eda0120ab478674d1da58d06431229ca13f6fcbf1c3dd3cfa7f64"
```
Comment (SMillerDev) [inline Formula/tfel.rb:10]: Don't edit this

File: Formula/tfel.rb [hunk 2]
```diff
@@ -17,16 +17,24 @@ class Tfel < Formula
 
   depends_on "cmake" => :build
   depends_on "gcc" => :build
-  depends_on "python@3.10" => :build
+  depends_on "boost-python3"
+  depends_on "python@3.10"
   fails_with gcc: "5"
 
+  patch do
+    url "https://github.com/thelfer/tfel/raw/fca68680750847fd56315664ecad19089bf9f0b7/patches/homebrew/tfel-4.0.0.patch"
```
Comment (SMillerDev) [inline Formula/tfel.rb:25]: Is this actually accepted upstream? Or is it just hosted there for Homebrew?

File: Formula/tfel.rb [hunk 3]
```diff
@@ -37,6 +45,8 @@ def install
       "-Denable-diana-fea=ON",
       "-Denable-ansys=ON",
       "-Denable-europlexus=ON",
+      "-Dpython-static-interpreter-workaround=ON",
```
Comment (SMillerDev) [inline Formula/tfel.rb:48]: What does this do?

File: Formula/tfel.rb (hunk 2 above)
Comment (maxence-wz) [reply Formula/tfel.rb:25]: Changes are being pushed into the development branch

Comment (maxence-wz) [reply Formula/tfel.rb:25]: @SMillerDev Is the patch will also be applied to the master branch with the `--head` option?

Comment (SMillerDev) [reply Formula/tfel.rb:25]: In the current form, yes. You can wrap it in a `stable` block when it's merged

File: Formula/tfel.rb (hunk 3 above)
Comment (SMillerDev) [reply Formula/tfel.rb:52]: Why is it actually linking against python?

File: Formula/tfel.rb (hunk 2 above)
Comment (maxence-wz) [reply Formula/tfel.rb]: Patch wrapped and modification merged here : https://github.com/thelfer/tfel/commit/f610e638fde512cc1d94127458c89297e0ddaab3

**Comment by SMillerDev** [comment]
It still needs a response to my question "Why is it actually linking against python?", requesting a new review won't change that.

**Comment by SMillerDev** [comment]
Please stop requesting new reviews if you don't address the existing ones.

**Comment by maxence-wz** [comment]
Hello @SMillerDev, sorry I don't know why my comments were in 'pending' status. You may not see them until now.

File: Formula/tfel.rb (hunk 3 above)
Comment (SMillerDev) [reply Formula/tfel.rb:52]: Are you linking against it because you want to embed the interpreter in the library?

Comment (maxence-wz) [reply Formula/tfel.rb:52]: @SMillerDev, no we don't want to do that, but maybe I don't fully understand the purpose of your question. Could you be more specific?

Comment (SMillerDev) [reply Formula/tfel.rb:52]: Okay, found a better explanation than I can give. Just want to make sure that it's the correct fix for the problem. https://blog.tim-smith.us/2015/09/python-extension-modules-os-x/ has more info on the problem.

Comment (maxence-wz) [reply Formula/tfel.rb:52]: That exactly the point of the fix.

BTW, thanks for pointing this blog post which highlights and explains the trouble that we had when porting TFEL/MFront to Mac Os. We probably will change our default policy in the future (regarding linking with -lpython) based on those guidelines, thus removing the need for this option. For now on, this is required for the current stable version.

**Review by SMillerDev — APPROVED** [review]
Thanks @maxence-wz  ! Without contributions like yours it'd be impossible to keep homebrew going with the high standards that users have come to expect from the project. You can feel good knowing that you've made the world a tiny bit better for homebrew users around the world! 👍 🎉

**Pull request closed without merging**